core/server/master/src/main/java/alluxio/master/file/meta/SimpleInodeLockList.java

```diff
@@ -56,43 +65,43 @@
    */
   public SimpleInodeLockList(InodeLockManager inodeLockManager) {
     mInodeLockManager = inodeLockManager;
-    mEntries = new ArrayList<>(INITIAL_CAPACITY);
-    mFirstWriteLockIndex = 0;
+    mInodes = new ArrayList<>(INITIAL_CAPACITY);
+    mLocks = new ArrayList<>(INITIAL_CAPACITY);
```

Comment:
Is the initial capacity an issue for memory consumption? This change effectively made the initial size twice as large

Reply:
I changed to reserve less capacity.

Reply:
2 is too few as it only covers locking a hierarchy of 1 level deep.
Since this struct grows progressively with every lock and we remove only from tail, I believe LinkedList could be of use here instead of ArrayList. WDYT?

Reply:
My only concern of using LinkedList is that the implementation of `get` is O(n). But I checked that the only usage of `get` is to get the last or the second last element, which is O(1) in LinkedList. So I changed to LinkedList.